.github/workflows/tag-release.yml

```diff
@@ -1,36 +1,68 @@
-name: Create Releases on Tags
+name: Auto tag-release-publish
 
 on:
   push:
-    tags:
-      - 'v*'
+    branches:
+      - master
 
 jobs:
-  release:
+  tag:
+    name: Create tag for new version
     runs-on: ubuntu-latest
+    outputs:
+      tag_name: ${{ steps.create_new_tag.outputs.tag }}
     steps:
-      - name: Create GitHub release
-        uses: Roang-zero1/github-create-release-action@master
+      - uses: actions/checkout@v2
         with:
-          version_regex: ^v[[:digit:]]+\.[[:digit:]]+\.[[:digit:]]+
-          changelog_file: CHANGELOG.md
+          fetch-depth: 2
+      - uses: salsify/action-detect-and-tag-new-version@v2
+        id: create_new_tag
+
+  release:
+    name: Create release
+    runs-on: ubuntu-latest
+    needs: tag
+    if: needs.tag.outputs.tag_name
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/create-release@v1
+        id: create_release
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          tag_name: ${{ needs.tag.outputs.tag_name }}
+          release_name: ${{ needs.tag.outputs.tag_name }}
+          draft: false
+          prerelease: false
+
+  npm-publish:
+    name: Build and publish
+    runs-on: ubuntu-latest
+    needs: tag
+    if: needs.tag.outputs.tag_name
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: '14.x'
+          registry-url: 'https://registry.npmjs.org'
+      - run: yarn
+      - run: yarn build
```

Comment:
This task (i.e., `yarn build`) is also run inside the docker container in the `docker` job. It might not be a priority now, but you could theoretically fetch the built artifacts and push those to `npm` as opposed to building them twice. That would lower the total time this workflow takes.

Reply:
ooo this is a good idea! i'll make an tech debt or onboarding ticket for this, best place for that @karlfloersch ?

Reply:
made it here: https://github.com/ethereum-optimism/roadmap/issues/725

Reply:
Good job making a ticket for it